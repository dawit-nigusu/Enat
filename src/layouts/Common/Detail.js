import React from "react";
import {Image, ImageBackground, StyleSheet} from "react-native";
import {Button, Icon, View, Text, Left, Right} from "native-base";
import {Row, Col} from "react-native-easy-grid";

const ReactNative = require("react-native");
const {Dimensions} = ReactNative;
const deviceHeight = Dimensions.get("window").height;
const deviceWidth = Dimensions.get("window").width;

const cover = require("../../../assets/drawer-cover.png");

const DetailHeader = (props) => {
    return (
        <View style={{
            backgroundColor: '#eb3a7c',
            paddingTop: 5,
            paddingBottom: 5,
            height: 50,
            flexDirection: 'row',
            justifyContent: 'flex-start'
        }}>
                <Button
                    transparent
                    style={{ borderWidth: 0, height: 40 }}
                    onPress={props.onPress}>
                    <Icon style={{color: "#fff"}} name="chevron-thin-left" type='Entypo'/>
                </Button>
            <View sytle={{
                flex: 0.8,
                marginRight: 15,
            }}>
                <Text style={{
                    color: '#ffffff',
                    height: 40,
                    width: deviceWidth * 0.8,
                    fontSize: 16,
                    fontWeight: 'normal',
                    textAlignVertical: 'center',
                    textAlign:'justify',
                    flexWrap: 'wrap',
                }} numberOfLines={2}>{props.text}</Text>
            </View>

        </View>
    )
};

const TabHeader = (props) => {
    let prev = null;
    let next = null;
    if(props.prev === true) {
        prev = (<Button transparent
                                          style={{
                                              borderWidth: 0,
                                              height: 40
                                          }}
                                          onPress={props.onPressPrev}
            >
                <Icon style={{color: "#c31e5c"}} name="chevron-small-left" type='Entypo'/>
            </Button>);
    }
    if(props.next === true) {
        next = (<Button transparent
                style={{
                    borderWidth: 0,
                    height: 40
                }}
                onPress={props.onPressNext}
        >
            <Icon style={{color: "#c31e5c"}} name="chevron-small-right" type='Entypo'/>
        </Button>);
    }
    return (
        <View style={[styles.tabHeader]}>
            <Left>
                {prev}
            </Left>

            <View sytle={{
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
                paddingRight: 5,
            }}>
                <Text style={styles.tabHeaderText}>{props.tabname}</Text>
            </View>
            <Right>
                {next}
            </Right>

        </View>
    )
};

const TitleHeader = (props) => {
    return (
        <View style={styles.titleContainer}>
            <Text style={styles.title}>{props.text}</Text>
        </View>
    )
};

const Banner = (props) => {
    // TODO::fetch with props.bannerId from Asyncstorage
    // return (
    //     <Row>
    //         <Image
    //             source={cover}
    //             style={{
    //                 flex: 1,
    //                 width: null,
    //                 height: 90,
    //                 marginTop: 5,
    //                 marginBottom: 5,
    //             }}/>
    //     </Row>
    // )
    return (
        <Image
                source={{
                    uri: 'http://enat.hlmikhatech.com/banners/banner.jpg',
                    cache: 'force-cache',
                }}
                style={[{
                    flex: 1,
                    width: '100%',
                    height: 90,
                    marginTop: 5,
                }]}/>
    );

};

const SubTitleHeader = (props) => {
    return (
        <View style={styles.subtitleContainer}>
            <Text style={styles.subtitle}>{props.text}</Text>
        </View>
    )
};

const Paragraph = (props) => {
    return (
        <Row style={[styles.body]}>
            <Text style={[styles.text, {
                marginBottom: 0
            }]}>
                {props.text}
            </Text>
        </Row>
    )
};

const Bold = (props) => {
    return (
        <View>
            <Text style={{fontWeight:'bold'}}>
                {props.text}
            </Text>
        </View>
    )
};

const BulletinItem = (props) => {
    return (
        <Row style={styles.body}>
            <Text style={styles.bullet}> {'\u2022'} </Text>
            <Text style={styles.bulletText}>
                {props.text}
            </Text>
        </Row>
    )
};

const Bulletin = (props) => {
    return (
        <View>
            {props.list.map((item) => {
                return (
                    <BulletinItem key={item.id} text={item.text}/>
                );
            })}
        </View>
    )
};

const ImageJumbotron = (props) => {
    return (
        <Row style={styles.mb10}>
            <ImageBackground
                source={props.image}
                style={styles.image}

            >
            </ImageBackground>
        </Row>
    )
};

const Thumbnail = (props) => {
    return (
        <Row style={[styles.body, styles.mb10]}>
            <Col style={{
                flex:3
            }}>
                <ImageBackground
                    source={props.primaryImage}
                    style={[styles.primaryImage, styles.bottom]}
                >
                </ImageBackground>
            </Col>
            <Col style={{
                paddingTop: 5,
                paddingLeft: 25,
                flex:2
            }}>
                <Text style={{
                    fontSize: 12,
                    lineHeight:25,
                }}>{props.primaryText}</Text>
                <Text style={{
                    fontSize: 12,
                    lineHeight:20,
                    color:'rgb(194, 24, 91)'
                }}>{props.secondaryText}</Text>
                <Text style={{
                    fontSize: 12,
                    lineHeight:20,
                }}>{props.tertiaryText}</Text>
                {props.secondaryImage !== undefined ? <ImageBackground
                    source={props.secondaryImage}
                    style={[styles.secondaryImage]}/> : null}

            </Col>
        </Row>
    )
};

const styles = StyleSheet.create({
    mainHeader: {
        backgroundColor: '#eb3a7c',
        paddingTop: 5,
        paddingBottom: 5,
        paddingRight: 5,
        height: 50,
        flexDirection: 'row',
        justifyContent: 'space-around'
    },
    mainHeaderText: {
        color: '#ffffff',
        marginRight: 10,
        height: 40,
        alignSelf: 'flex-start',
        fontSize: 16,
        fontWeight: 'normal',
        textAlignVertical: 'center',
        flexWrap: 'wrap',
        flex:0.9,
    },
    tabHeader: {
        backgroundColor: '#fdecf3',
        paddingTop: 5,
        paddingBottom: 5,
        paddingRight: 5,
        height: 50,
        flexDirection: 'row',
        alignItems: 'center',
    },
    tabHeaderText: {
        color: '#c31e5c',
        paddingRight: 5,
        height: 20,
        alignSelf: 'center',
        fontSize: 14,
        fontWeight: 'normal',
        textAlignVertical: 'center',
        flexWrap: 'wrap',
    },
    body: {
        marginLeft: 10,
        marginRight: 10,
        marginBottom: 10,
    },
    text: {
        bottom: 6,
        marginTop: 5,
        alignSelf: "flex-start",
        marginBottom: 7,
        marginRight: 10,
        fontSize: 13,
        letterSpacing: 1,
        color: "#444444",
        lineHeight: 18,
    },
    mb10: {
        marginBottom: 10
    },
    stretch: {
        flex: 1,
    },
    titleContainer: {
        backgroundColor: '#fdecf3',
        flex: 1,
        marginBottom: 10,
        flexDirection: 'row',
        justifyContent: 'flex-start',
    },
    title: {
        backgroundColor: '#fdecf3',
        color: '#be1f57',
        fontWeight: 'bold',
        fontSize: 17,
        padding: 10,
    },
    subtitleContainer: {
        backgroundColor: '#fff',
        flex: 1,
        marginBottom: 10,
        flexDirection: 'row',
        justifyContent: 'flex-start',
    },
    subtitle: {
        backgroundColor: '#fff',
        color: '#be1f57',
        fontSize: 16,
        padding: 10,
    },
    bullet: {
        color: "#e82064"
    },
    bulletText: {
        marginLeft: 5,
        marginRight: 10,
        fontSize: 13,
        letterSpacing: 1,
        color: "#686463",
        lineHeight: 18,
    },
    image: {
        width: deviceWidth,
        height: (167 * deviceWidth)/375,
    },
    primaryImage: {
        width: deviceWidth * 0.55,
        height: 120,
        justifyContent: 'flex-end'
    },
    secondaryImage: {
        flex: 1,
        width: 76,
        height: 60,

        justifyContent: 'flex-end'
    }

});

export {
    DetailHeader,
    Paragraph,
    Bulletin,
    ImageJumbotron,
    TitleHeader,
    SubTitleHeader,
    Thumbnail,
    Banner,
    TabHeader,
    Bold
};
