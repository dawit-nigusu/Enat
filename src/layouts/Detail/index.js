import React, {Component} from "react";
import {Container, Content } from "native-base";
import Header from "../Common/Header";
import {
    DetailHeader, Paragraph, Bulletin, ImageJumbotron,
    TitleHeader, SubTitleHeader, Thumbnail, Banner, Bold,
} from "../Common/Detail";
import Tabs from "./tab";
import KickCounter from "../KickCounter";
import Vaccination from "../Vaccination";
import Symptoms from "../Symptoms";
import Contacts from "../Calls";
import BabyName from "../BabyName"
import { BackHandler } from "react-native";

class Detail extends Component {

    constructor(props){
      super(props);
      this.backButtonClick = this.backButtonClick.bind(this);
    }

    componentWillMount() {
      BackHandler.addEventListener('hardwareBackPress', this.backButtonClick);
    }

    componentWillUnmount(){
      BackHandler.removeEventListener('hardwareBackPress', this.backButtonClick);
    }

    backButtonClick = () => {
      if(this.props.navigation && this.props.navigation.goBack){
        this.props.navigation.navigate(this.props.navigation.getParam('back', 'Home'));
        return true;
      }
      return false;
    }

    render() {
        const navigation = this.props.navigation;
        const header = navigation.getParam('header', []);
        const data = navigation.getParam('data', []);
        const route = navigation.getParam('back', 'Home');
        return (
            <Container style={{
                backgroundColor: "#FFF"
            }}>
                <Header />
                <DetailHeader text={header}
                              onPress={() => navigation.navigate(route)}/>
                <Content>
                    {data.map((dataInfo) => {
                        if (dataInfo.type === 'title') {
                            return (
                                <TitleHeader
                                    key={dataInfo.id}
                                    text={dataInfo.text}/>
                            );
                        }
                        else if (dataInfo.type === 'subtitle') {
                            return (
                                <SubTitleHeader
                                    key={dataInfo.id}
                                    text={dataInfo.text}/>
                            )
                        }
                        else if (dataInfo.type === 'text') {
                            return (
                                <Paragraph
                                    key={dataInfo.id}
                                    text={dataInfo.text}/>
                            );
                        }
                        else if (dataInfo.type === 'bold') {
                            return (
                                <Bold
                                    key={dataInfo.id}
                                    text={dataInfo.text}/>
                            );
                        }
                        else if (dataInfo.type === 'banner') {
                            return (
                                <Banner
                                    key={dataInfo.id}
                                    bannerid={dataInfo.bannerId}/>
                            );
                        }
                        else if (dataInfo.type === 'jumbotron') {
                            return (
                                <ImageJumbotron
                                    key={dataInfo.id}
                                    image={dataInfo.image}/>
                            );
                        }
                        else if (dataInfo.type === 'thumbnail') {
                            return (
                                <Thumbnail
                                    key={dataInfo.id}
                                    primaryImage={dataInfo.primaryImage}
                                    primaryText={dataInfo.primaryText}
                                    secondaryText={dataInfo.secondaryText}
                                    tertiaryText={dataInfo.tertiaryText}
                                    secondaryImage={dataInfo.secondaryImage}
                                />
                            );
                        }
                        else if (dataInfo.type === 'bulletin') {
                            return (
                                <Bulletin
                                    key={dataInfo.id}
                                    list={dataInfo.list}/>
                            );
                        }
                        else if (dataInfo.type === 'tabs') {
                            return (
                                <Tabs
                                    key={dataInfo.id}
                                    tabs={dataInfo.tabs}
                                    image={dataInfo.image}/>);
                        }
                        else if (dataInfo.type === 'kickcounter') {
                            return (
                                <KickCounter
                                    key={dataInfo.id}/>
                            );
                        }
                        else if (dataInfo.type === 'vaccination') {
                            return (
                                <Vaccination
                                    key={dataInfo.id}
                                    image={dataInfo.image}/>
                            );
                        }
                        else if (dataInfo.type === 'symptoms') {
                            return (
                                <Symptoms
                                    key={dataInfo.id}
                                    image={dataInfo.image}/>
                            );
                        }
                        else if (dataInfo.type === 'contacts') {
                            return (
                                <Contacts
                                    key={dataInfo.id}
                                    image={dataInfo.image}/>
                            );
                        }
                        else if (dataInfo.type === 'babyname') {
                            return (
                                <BabyName
                                    key={dataInfo.id}
                                    image={dataInfo.image}/>
                            );
                        }
                    })}
                </Content>
            </Container>
        );
    }
}

export default Detail;
