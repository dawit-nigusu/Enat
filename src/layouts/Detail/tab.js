import React, {Component} from "react";
import { Container, Content } from "native-base";
import Banner from "../Common/Banner";
import {
    Bulletin,
    ImageJumbotron,
    Paragraph,
    SubTitleHeader,
    TabHeader,
    Thumbnail,
    TitleHeader
} from "../Common/Detail";
import {Row, Col} from "react-native-easy-grid";
import { Image, View } from "react-native";

const ReactNative = require("react-native");
const {Dimensions} = ReactNative;
const deviceHeight = Dimensions.get("window").height;

class Tabs extends Component {
    constructor(props) {
        super(props);

        let _tabs = this.props.tabs;
        let _currentTab = {
            name: '',
            data: []
        };
        let _next = false;
        if (_tabs.length > 0) {
            _currentTab = _tabs[0];
        }
        if (_tabs.length > 1) {
            _next = true;
        }

        this.state = {
            prev: false,
            tabIndex: 0,
            tabs: _tabs,
            currentTab: _currentTab,
            next: _next,
        };
    }

    _handleOnPressNext = () => {
        let _tabIndex = this.state.tabIndex;
        if (_tabIndex < this.state.tabs.length - 1 ) {
            _tabIndex = _tabIndex + 1;
            this.setState({
                tabIndex: _tabIndex,
                currentTab: this.state.tabs[_tabIndex],
                prev: true,
            });
        }
        if (_tabIndex < this.state.tabs.length - 1 ) {
            this.setState({next: true});
        } else {
            this.setState({next: false});
        }
    };

    _handleOnPressPrev = () => {
        let _tabIndex = this.state.tabIndex;
        if (_tabIndex > 0 ) {
            _tabIndex = _tabIndex - 1;
            this.setState({
                tabIndex: _tabIndex,
                currentTab: this.state.tabs[_tabIndex],
                next: true
            });
        }
        if (_tabIndex > 0 ) {
            this.setState({prev: true});
        } else {
            this.setState({prev: false});
        }
    };

    render() {
        return (
            <Col>
                <TabHeader
                    prev={this.state.prev}
                    onPressPrev={this._handleOnPressPrev}
                    tabname={this.state.currentTab.name}
                    next={this.state.next}
                    onPressNext={this._handleOnPressNext}
                />
                {
                    this.props.image !== undefined ? <ImageJumbotron
                        image={this.props.image}/> : null
                }
                <View>
                    {this.state.currentTab.data.map((dataInfo) => {
                        if (dataInfo.type === 'title') {
                            return (
                                <TitleHeader
                                    key={dataInfo.id}
                                    text={dataInfo.text}/>
                            );
                        }
                        else if (dataInfo.type === 'subtitle') {
                            return (
                                <SubTitleHeader
                                    key={dataInfo.id}
                                    text={dataInfo.text}/>
                            );
                        }
                        else if (dataInfo.type === 'text') {
                            return (
                                <Paragraph
                                    key={dataInfo.id}
                                    text={dataInfo.text}/>
                            );
                        }
                        else if (dataInfo.type === 'banner') {
                            return (
                                <Banner
                                    key={dataInfo.id}
                                    image={dataInfo.image}/>
                            );
                        }
                        else if (dataInfo.type === 'jumbotron') {
                            return (
                                <ImageJumbotron
                                    key={dataInfo.id}
                                    image={dataInfo.image}/>
                            );
                        }
                        else if (dataInfo.type === 'fullscreenImage') {
                            return (
                                <Image
                                    key={dataInfo.id}
                                    source={dataInfo.image}
                                    style={{
                                        flex:1,
                                        width: null,
                                        height: deviceHeight * 0.8,
                                    }}/>
                            )
                        }
                        else if (dataInfo.type === 'thumbnail') {
                            return (
                                <Thumbnail
                                    key={dataInfo.id}
                                    primaryImage={dataInfo.primaryImage}
                                    primaryText={dataInfo.primaryText}
                                    secondaryText={dataInfo.secondaryText}
                                    tertiaryText={dataInfo.tertiaryText}
                                    secondaryImage={dataInfo.secondaryImage}
                                />
                            );
                        }
                        else if (dataInfo.type === 'bulletin') {
                            return (
                                <Bulletin
                                    key={dataInfo.id}
                                    list={dataInfo.list}/>
                            );
                        }
                    })}
                </View>
            </Col>
        );
    }
}

export default Tabs;
