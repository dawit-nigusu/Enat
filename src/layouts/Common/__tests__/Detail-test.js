import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { SubTitleHeader, Paragraph, Bold, TitleHeader } from "../Detail";

describe('Detail page components test', () => {
    it('renders title component', () => {
        expect(renderer.create(
            <TitleHeader text={'hello'}/>
        ).toJSON()).toMatchSnapshot();
    });

    it('renders SubTitle component', () => {
        const subtitle = renderer.create(
            <SubTitleHeader text={'hello'}/>
        ).toJSON();
        expect(subtitle).toMatchSnapshot();
    });

    it('renders paragraph component', () => {
        const paragraph = renderer.create(
            <Paragraph text={'hello'}/>
        ).toJSON();
        expect(paragraph).toMatchSnapshot();
    });

    it('renders bold component', () => {
        expect(renderer.create(
            <Bold text={'hello'}/>
        ).toJSON()).toMatchSnapshot();
    });
});
