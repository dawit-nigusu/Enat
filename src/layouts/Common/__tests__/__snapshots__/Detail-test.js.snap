// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail page components test renders SubTitle component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 10,
      },
    ]
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "backgroundColor": "#fff",
          "color": "#be1f57",
          "fontSize": 16,
          "padding": 10,
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  >
    hello
  </Text>
</View>
`;

exports[`Detail page components test renders bold component 1`] = `
<View
  style={Object {}}
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  >
    hello
  </Text>
</View>
`;

exports[`Detail page components test renders paragraph component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "alignSelf": "flex-start",
          "bottom": 6,
          "color": "#444444",
          "fontSize": 13,
          "letterSpacing": 1,
          "marginBottom": 7,
          "marginRight": 10,
          "marginTop": 5,
        },
        Object {
          "marginBottom": 0,
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  >
    hello
  </Text>
</View>
`;

exports[`Detail page components test renders title component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "#fdecf3",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 10,
      },
    ]
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "backgroundColor": "#fdecf3",
          "color": "#be1f57",
          "fontSize": 17,
          "fontWeight": "bold",
          "padding": 10,
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  >
    hello
  </Text>
</View>
`;
